app/scripts/modules/core/src/pipeline/executions/execution/ExecutionMarker.tsx

```diff
@@ -81,15 +84,26 @@ export class ExecutionMarker extends React.Component<IExecutionMarkerProps, IExe
     });
   };
 
+  private stageStatus = (stageStatus: string) => {
+    if (stageStatus === 'running') {
+      let temp = this.props.stage.stages.filter(
+        (stage) => stage.status.toLowerCase() === 'running' && stage.type === 'pipeline' && !isEmpty(stage.others),
+      );
+      if (!isEmpty(temp)) return 'waiting';
+    }
+    return stageStatus;
+  };
+
   public render() {
     const { stage, application, execution, active, previousStageActive, width } = this.props;
     const stageType = (stage.activeStageType || stage.type).toLowerCase(); // support groups
+    const PIPELINE_STATUS = this.stageStatus(stage.status.toLowerCase());
```

Comment:
nit: Call this `pipelineStatus` since it isn't a constant. Usually we reserve all caps for constants.

Reply:
changed PIPELINE_STATUS to pipelineStatus